src/metax_api/tests/api/rest/base/views/datasets/write.py

```diff
@@ -343,6 +343,30 @@ def test_anonymous_user_cannot_update_draft_datasets(self):
             response = update_request('/rest/datasets', data, format="json")
             self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN, response.status_code)
 
+    ###
+    # Tests for deleting drafts
+    ###
+
+    def test_draft_is_permanently_deleted_by_service_user(self):
+        '''Draft datasets should be permanently deleted from the database.
+        Only the dataset owner is able to delete draft datasets.'''
+        # Set service-user
+        self._use_http_authorization(method='basic', username='metax')
+
+        for cr_id in (2, 3):
+            response = self.client.delete('/rest/datasets/%d' % cr_id)
+            self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT, response.data)
+
+    @responses.activate
+    def test_draft_is_permanently_deleted_by_enduser(self):
+        # Set end user
+        self._use_http_authorization(method='bearer', token=self.token)
+
+        response = self.client.delete('/rest/datasets/2')
+        self.assertEqual(response.status_code, status.HTTP_204_NO_CONTENT, response.status_code)
+        response = self.client.delete('/rest/datasets/3')
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND, response.status_code)
```

Comment:
In tests where drafts are deleted and dataset is permanently deleted, you should check that the dataset was really removed from the db. You can do that with ``cr_exists = CatalogRecord.objects_unfiltered.filter(pk=1).exists()`` and assert that ``cr_exists`` is ``False``.

The following could be used too: ``num_found = CatalogRecord.objects_unfiltered.filter(pk__in=[1, 2]).count()``, and ``num_found`` should then be zero obviously.

Reply:
Ok added asserts to check this